### ISSUE TITLE
Add collect_stats option to losses and miners

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#40]
Some of the losses and miners do additional computations for the purpose of experiment analysis. For example, ArcFaceLoss computes ```avg_angle``` which the user can log on tensorboard etc. 

However this is not actually needed to compute the loss, so it would be nice to have a collect_stats option, which when set to False, will skip computations that aren't necessary for computing the loss.

All stats computations should also be wrapped with torch.no_grad()

**Comment by KevinMusgrave** [comment]
Added to v0.9.90. See [the documentation](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#basemetriclossfunction)
